wordpress-mobile/WordPress-Android
# Update reader site header UI

This PR updates the Reader Site Header UI according to the design concepts provided.  It addresses this [task](https://github.com/orgs/wordpress-mobile/projects/63#card-42500990)

There were a couple of gotchas that were uncovered while addressing this PR.
The `ReaderSiteHeaderView` has support for viewing by feed or blog. In both cases, ReaderBlog data is gathered from the ReaderBlogTable, but feed/blog requests use different endpoints to fetch the data from the server. The icon image url is not available in the `/read/feed/{feedId}` endpoint, so I had to add an extra server call to `/read/sites/{blogId}` to get the url when the request is for a feed. 

In addition, when working within the context of a feed, the blogId is set to the feedId value. This means that I had to first query the post table for the "blogId" for the feed. This was necessary to make the correct call to the server for the image. 

I have added comments in the code to highlight why changes were made. Please let me know if they are not clear. 

Light | Dark
--------|------- 
<kbd><img src="https://user-images.githubusercontent.com/506707/89952747-cb29a600-dbfb-11ea-8e94-b8a6ab4b6b68.png" width=320></kbd> | <kbd><img src="https://user-images.githubusercontent.com/506707/89952764-d1b81d80-dbfb-11ea-8c33-ee6a4228d457.png" width=320></kbd>

The new following functionality for non logged in users is behind a FF for RI improvements, so the logged out functionality would look like the following if this PR was released without a FF.
<kbd><img src="https://user-images.githubusercontent.com/506707/90009157-8ccbcf80-dc6b-11ea-9491-11474035cc80.png" width=320></kbd>

To test
1. Go to reader
2. Tap on a site name in a post card
3. See the new site header

PR submission checklist:
